packages/mui-system/src/colorManipulator.js

```diff
@@ -274,11 +323,16 @@ export function darken(color, coefficient) {
 
   if (color.type.indexOf('hsl') !== -1) {
     color.values[2] *= 1 - coefficient;
-  } else if (color.type.indexOf('rgb') !== -1 || color.type.indexOf('color') !== -1) {
+  } else if (color.type.indexOf('rgb') !== -1) {
+    color = decomposeColor(rgbToHsl(color));
+    color.values[2] *= 1 - coefficient;
+    color = decomposeColor(hslToRgb(color));
```

Comment:
I couldn't find a more efficient way to do this. It does seem inefficient to continuously decompose colors, but it was the cleanest way to keep color formats organized.